pkg/provider/resource_cluster.go

```diff
@@ -652,11 +675,42 @@ func makeClusterForCreate(clusterResource ClusterResourceModel) models.Cluster {
 		}
 	}
 
+	if strings.Contains(clusterResource.CloudProvider.ValueString(), "bah") {
```

Comment:
Have this in another function maybe called `setBahSettingsToClusterAs` it would also be easier to test this function in isolation